packages/eventual-send/src/index.d.ts

```diff
@@ -18,6 +18,20 @@ export type EOnly<T> = T extends (...args: infer P) => infer R
     >
   : ERef<T>;
 
+type FilteredKeys<T, U> = {
+  [P in keyof T]: T[P] extends U ? P : never;
+}[keyof T];
+
+type DataOnly<T> = Omit<T, FilteredKeys<T, Function>>;
+type FunctionOnly<T> = Pick<T, FilteredKeys<T, Function>> &
+  (T extends Function ? (...args: Parameters<T>) => ReturnType<T> : {});
+
+interface Remotable<T> {
+  __Remote__: T;
+}
+type Remote<Primary, Local = DataOnly<Primary>> = ERef<
+  Local & Remotable<Primary>
+>;
 type Unpromise<T> = T extends ERef<infer U> ? U : T;
```

Comment:
If we're deprecating `Unpromise` in favor of `Awaited`, should we formally deprecate `Unpromise` here? Together with a note to use `Awaited` instead?

Reply:
I do that in the next PR (#1102).  Nobody else was using `Unpromise`, so I didn't leave a note.